src/bin/pgcopydb/table-data.c

```diff
@@ -1161,6 +1161,9 @@ copydb_table_create_lockfile(CopyDataSpec *specs,
 		}
 
 		args->truncate = granted;
+
+		/* cannot perform COPY FREEZE if the table was not created or truncated in the current subtransaction */
+		args->freeze &= granted;
```

Comment:
Can we do this from [pgsql.c](https://github.com/dimitri/pgcopydb/blob/4f41dd430c8160876b89c26a99d81fc068d54c99/src/bin/pgcopydb/pgsql.c#L2737)? That would be a common place and applies to all usage of pg_copy?

Reply:
Although we can do this from pgsql.c, I believe this is a better solution. The function that builds the arguments should decide on the fields, and pgsql.c should execute them.

The comment block above suggests that we should decide whether we can benefit from `with (freeze)` in this file.

FWIW, the following patch seems to solve the issue as well, but I can not say that I like it more than this PR here.

```diff
diff --git a/src/bin/pgcopydb/pgsql.c b/src/bin/pgcopydb/pgsql.c
index 09c362c..c45229c 100644
--- a/src/bin/pgcopydb/pgsql.c
+++ b/src/bin/pgcopydb/pgsql.c
@@ -3106,7 +3106,7 @@ pg_copy_send_query(PGSQL *pgsql, CopyArgs *args, ExecStatusType status)
             appendPQExpBuffer(sql, "copy %s from stdin", args->dstQname);
         }

-        if (args->freeze)
+        if (args->freeze && args->truncate)
         {
             appendPQExpBuffer(sql, " with (freeze)");
         }
```